cmd/snap-confine/user-support.c

```diff
@@ -37,6 +38,11 @@ void setup_user_data(void)
 
 	debug("creating user data directory: %s", user_data);
 	if (sc_nonfatal_mkpath(user_data, 0755) < 0) {
+		if (errno == EROFS && !sc_startswith(user_data, "/home/")) {
+				// clear errno or it will be displayed in die()
+				errno = 0;
+				die("Sorry, cannot use home directories outside /home.\nSee https://forum.snapcraft.io/t/11209 for details.");
```

Comment:
Wonder if @degville has some ideas about tweaking the message.

Reply:
I'd probably slightly tweak:
`Sorry, home directories outside of /home are not currently supported. \nSee https://forum.snapcraft.io/t/11209 for details.`

I think it's worth updating the forum post too, including the bind mount workaround and adding it to the docs - I'm obviously happy to do this if it helps.

Reply:
Thanks, I updated the message. Agreed on the forum post - its a skeleton at this point mostly because I was too lazy^Wbusy to flesh it out. If you want to have a stab I would be very happy!